debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/PostgresConnectorIT.java

```diff
@@ -1018,7 +1017,8 @@ public void shouldNotProduceEventsWithInitialOnlySnapshot() throws InterruptedEx
         assertConnectorIsRunning();
 
         // check the records from the snapshot
-        assertRecordsFromSnapshot(2, 1, 1);
+        // Add extra +2 for the heartbeat records
+        assertRecordsFromSnapshot(2+2, 1, 1);
```

Comment:
My understanding is that the heartbeat records get sent every 1 second till we receive the acknowledgement, so in case of this test here, are we absolutely sure that once we have received 2 heartbeat records, it's alright to go ahead and call this a passed test assuming all the records have been consumed?

Reply:
For the UT, offset.flush.interval is set to 0 - this causes every event to commit offset. Thus exactly 2 heartbeats will be received - one from RelationalSnapshotChangeEventSource and one from waiting for the snapshot to streaming transition wait loop.